app/models/user.rb

```diff
@@ -16,6 +16,10 @@ def self.all_except(user)
     where.not(id:  ids += user.friends.ids)
   end
 
+  def active_for_authentication?
+    super && state == "active"
```

Comment:
what does super do here?

Reply:
It's a devise action, so to modify it I also have to use super keyword to use devise's action.